browser/src/layer/tile/CanvasSectionContainer.ts

```diff
@@ -576,6 +576,18 @@ class CanvasSectionContainer {
 		this.canvas.ontouchend = this.onTouchEnd.bind(this);
 		this.canvas.ontouchcancel = this.onTouchCancel.bind(this);
 
+		if (typeof Hammer !== 'undefined') {
+			// Mocha does not provide window but hammerjs requires it. As a workaround let's check if hammer exists before trying to create an instance
+			(new Hammer(this.canvas, {recognizers: [[Hammer.Press]]}))
```

Comment:
can we not use hammerjs?
there is `onLongPress` handler for canvas sections, can we use it?

Reply:
So, @eszkadev, I considered this... unfortunately I didn't manage to get a longpress working with this handler - it seemed it wasn't called. As it's not a DOM event and I can't see anywhere it's called except via `onMouseMove` (which isn't triggered on touch) I presumed it's not for this and wrote it with hammer instead... do you believe this to be an incorrect assumption?

We could write a standard Javascript solution, but I wonder if it's worth it... do you think there is significant benefit to keeping hammer.js away from here given that we depend on it in online already? I guess maybe for mocha, but beyond that I'm not so sure

Reply:
@gokaysatir is author of canvas section, maybe we need some trick to make it working. can you help here?

We tried to not rely on hammer here (in sections) as I know. Not introduce many new handlers fighting with each other, but rather create set of event handlers which propagate properly events. I think sections can be stacked on each other so it's important.

Reply:
What is the status here @Minion3665 @gokaysatir ? Do we have replacement for hammerjs ?

Reply:
There was a missing commit there. Now it's ready to be tested again.

Reply:
I tested it on real device and it seemed to work without hammer.js. Using mostly current implementation of canvasSectionContainer.